cvat-sdk/cvat_sdk/auto_annotation/driver.py

```diff
@@ -78,39 +126,119 @@ def _get_expected_function_output_type(self, fun_label, ds_label):
             )
         return fun_output_type
 
+    def _build_attribute_id_mapping(
+        self, fun_attr: models.IAttribute, ds_attr: models.IAttribute, attr_desc: str
+    ) -> _AttributeIdMapping:
+        # We could potentially be more lax with these checks. For example, we could permit
+        # fun_attr.values to be a subset of ds_attr.values. For simplicity though,
+        # we'll just use exact comparisons for now.
+        if ds_attr.input_type != fun_attr.input_type:
+            raise BadFunctionError(
+                f"{attr_desc} has input type {fun_attr.input_type!r} in the function,"
+                f" but {ds_attr.input_type!r} in the dataset"
+            )
+
+        if (
+            ds_attr.input_type.value not in {"text", "checkbox"}
+            and ds_attr.values != fun_attr.values
```

Comment:
Maybe relax the requirements a little and allow reordering?

Reply:
Yeah, okay. Updated.